File: Competetive programming/Python/Unit 3/Crypt_kicker_II.py
key = "the quick brown fox jumps over the lazy dog"

print("===== CRYPT KICKER II =====\n")

print("Enter encrypted sentence:")
cipher = input()

mapping = {}

for c, k in zip(cipher, key):

    if c != ' ':
        mapping[c] = k

decoded = ""

for ch in cipher:

    if ch == ' ':
        decoded += ' '
    else:
        decoded += mapping.get(ch, '?')

print("\nDecoded Message:")
print(decoded)

File: Competetive programming/C/Unit 3/Crypt_kicker_II.c
#include <stdio.h>
#include <string.h>
int main() {
    char encrypted[200];
    char key[] = "the quick brown fox jumps over the lazy dog";
    printf("Enter encrypted sentence:\n");
    fgets(encrypted, sizeof(encrypted), stdin);
    if(strlen(encrypted) - 1 != strlen(key)) {
        printf("No solution.\n");
        return 0;
    }
    char map[256] = {0};
    for(int i = 0; key[i]; i++) {
        if(key[i] == ' ') {
            if(encrypted[i] != ' ') {
                printf("No solution.\n");
                return 0;
            }
        }
        else {
            map[(int)encrypted[i]] = key[i];
        }
    }
    printf("\nDecoded Message:\n");
    for(int i = 0; encrypted[i]; i++) {

        if(encrypted[i] == ' ')
            printf(" ");
        else
            printf("%c", map[(int)encrypted[i]]);
    }

    return 0;
}